expecter_test.go

```diff
@@ -144,3 +144,29 @@ func TestMockPreload(t *testing.T) {
 		t.Error("In and out are not equal")
 	}
 }
+
+func TestMockPreloadMany2Many(t *testing.T) {
+	db, expect, err := gorm.NewDefaultExpecter()
+	defer func() {
+		db.Close()
+	}()
+
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	in := User{Id: 1}
+	languages := []Language{Language{Name: "ZH"}, Language{Name: "EN"}}
+	out := User{Id: 1, Languages: languages}
+
+	expect.Preload("Languages").Find(&in).Returns(out)
+	db.Preload("Languages").Find(&in)
+
+	if err := expect.AssertExpectations(); err != nil {
+		t.Error(err)
+	}
+
+	// if !reflect.DeepEqual(in, out) {
```

Comment:
@jinzhu For some reason, when returning mock rows from sqlmock, the values are not being scanned into the struct as expected. I can't find the reason why and need some help.